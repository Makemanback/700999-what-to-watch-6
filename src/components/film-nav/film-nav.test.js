import React from 'react';
import {Router} from 'react-router-dom';
import {render} from '@testing-library/react';
import {createMemoryHistory} from 'history';

import FilmNav from './film-nav';

const id = 1;
const Tabs = {
  OVERVIEW: `OVERVIEW`,
  DETAILS: `DETAILS`,
  REVIEWS: `REVIEWS`
};
const mockActiveTab = Tabs.OVERVIEW;

test(`Should FilmNav render correctly`, () => {
  const setActiveTab = jest.fn();
  const history = createMemoryHistory();

  const {container} = render(
      <Router history={history}>
        <FilmNav id={id} setActiveTab={setActiveTab} Tabs={Tabs} activeTab={mockActiveTab} />
      </Router>
  );

  expect(container).toMatchSnapshot();
});

