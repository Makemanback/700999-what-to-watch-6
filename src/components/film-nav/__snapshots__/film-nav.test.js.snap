// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should FilmNav render correctly 1`] = `
<div>
  <nav
    class="movie-nav movie-card__nav"
  >
    <ul
      class="movie-nav__list"
    >
      <li
        class="movie-nav__item movie-nav__item--active"
      >
        <a
          class="movie-nav__link movie-nav__item--active"
          href="/films/1"
        >
          Overview
        </a>
      </li>
      <li
        class="movie-nav__item"
      >
        <a
          class="movie-nav__link"
          href="/films/1"
        >
          Details
        </a>
      </li>
      <li
        class="movie-nav__item"
      >
        <a
          class="movie-nav__link"
          href="/films/1"
        >
          Reviews
        </a>
      </li>
    </ul>
  </nav>
</div>
`;
